### PR TITLE
Use less `#if` to handle `HC_NOZLIB` also fix `HC_NOWEBSOCKET`

### PR DESCRIPTION
**Pull request by lubeltra**

### CODE REVIEW

File: Build/libHttpClient.Common/libHttpClient.Common.vcxitems [hunk 1]
```diff
@@ -27,6 +27,8 @@
     <ClInclude Include="$(MSBuildThisFileDirectory)..\..\Include\httpClient\mock.h" />
     <ClInclude Include="$(MSBuildThisFileDirectory)..\..\Include\httpClient\pal.h" />
     <ClInclude Include="$(MSBuildThisFileDirectory)..\..\Include\httpClient\trace.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)..\..\Include\httpClient\types_android.h" />
```
Comment (jplafonta) [inline Build/libHttpClient.Common/libHttpClient.Common.vcxitems:30]: > <ClInclude Include="$(MSBuildThisFileDirectory)..\..\Include\httpClient\types_android.h" /> [](http://example.com/codeflow?start=3&length=92)

Should this be added here?

File: Include/httpClient/config.h [hunk 2]
```diff
@@ -6,7 +6,6 @@
 #define HC_PLATFORM_UWP 2
 #define HC_PLATFORM_XDK 3
 #define HC_PLATFORM_GDK 4
-#define HC_PLATFORM_GRTS 5
```
Comment (jplafonta) [inline Include/httpClient/config.h:9]: >HC_PLATFORM_GRTS [](http://example.com/codeflow?start=8&length=16)

is this not needed for backward compat anymore?

File: Source/HTTP/compression.cpp [hunk 3]
```diff
@@ -2,22 +2,24 @@
...
 #define CHUNK 16384
 #define WINDOWBITS 15
 #define GZIP_ENCODING 16
 
 NAMESPACE_XBOX_HTTP_CLIENT_BEGIN
 
-Compression::Compression()
+bool Compression::Available() noexcept
 {
+    return false;
```
Comment (jplafonta) [inline Source/HTTP/compression.cpp:19]: >false [](http://example.com/codeflow?start=11&length=5)

true?

File: Source/HTTP/httpcall_request.cpp [hunk 4]
```diff
@@ -108,6 +107,11 @@ try
     }
     RETURN_IF_PERFORM_CALLED(call);
 
+    if (!Compression::Available() && level != HCCompressionLevel::None)
+    {
+        return E_NOT_SUPPORTED;
+    }
+
```
Comment (jplafonta) [inline Source/HTTP/httpcall_request.cpp:114]: Any concern about this being a runtime failure as opposed to knowing its not supported at compile time?

File: Build/libHttpClient.Common/libHttpClient.Common.vcxitems (hunk 1 above)
Comment (lubeltra) [reply Build/libHttpClient.Common/libHttpClient.Common.vcxitems:30]: you are right, it's redundant

File: Include/httpClient/config.h (hunk 2 above)
Comment (lubeltra) [reply Include/httpClient/config.h:9]: we only ever used this internally in grts itself and we do not use it anymore

File: Source/HTTP/compression.cpp (hunk 3 above)
Comment (lubeltra) [reply Source/HTTP/compression.cpp:19]: I am a derp

File: Source/HTTP/httpcall_request.cpp (hunk 4 above)
Comment (lubeltra) [reply Source/HTTP/httpcall_request.cpp:114]: As discussed offline, the client should be free to just ignore the error, in the end they still want the request to be made compression or not.

**Pull request merged**